src/mono/wasm/runtime/library_mono.js

```diff
@@ -1135,6 +1135,38 @@ var MonoSupportLib = {
 			return res;
 		},
 
+		_get_shard_name: function (fileList) {
+			var name_lengths = fileList.map(x => x.split("_").length);
+			var shard_ix = name_lengths.indexOf(Math.max(...name_lengths));
```

Comment:
The way this is written makes it very hard to tell what it's doing.  There is nothing wrong with this code and it obviously works but it would be great to see a brief comment explaining why this index is calculated the way it is